### PR TITLE
Partitioning Historic Orders Data

### PR DESCRIPTION
**Pull request by italo-batista**
As title says

### CODE REVIEW

File: projects/orders-partition-historic/script-submit/config.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+         "spark.executor.heartbeatInterval":"60s",
+         "spark.rdd.compress":"true",
+         "spark.network.timeout":"800s",
+         "spark.executor.cores":"3",
+         "spark.memory.storageFraction":"0.30",
+         "spark.shuffle.spill.compress":"true",
+         "spark.shuffle.compress":"true",
+         "spark.storage.level":"MEMORY_AND_DISK_SER",
+         "spark.default.parallelism":"84",
+         "spark.memory.fraction":"0.80",
```
Comment (wellfabr) [inline projects/orders-partition-historic/script-submit/config.json]: Keep in mind that in general, you must keep the default setting for spark.memory.fraction. Increasing this value you may also increase the chance to have heap out of memory errors.

File: projects/orders-partition-historic/script-submit/config.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+         "spark.executor.extraJavaOptions":"-XX:+UseG1GC -XX:+UnlockDiagnosticVMOptions -XX:+G1SummarizeConcMark -XX:InitiatingHeapOccupancyPercent=35 -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:OnOutOfMemoryError='kill -9 %p'",
```
Comment (wellfabr) [inline projects/orders-partition-historic/script-submit/config.json]: From the experience I had with UseParallelGC you will get better stability, the trade-off is sometimes you may have some major GC pauses. Good for jobs that are not so much timing sensitive.

UseG1GC is where you are running a timing-sensitive jobs (like our Vtex Pulse). The trade-off is that you may get slight lower performance and maybe worst instability for heavy memory jobs (the GC may not be able to keep it)

File: projects/orders-partition-historic/script-submit/config.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+         "spark.executor.instances":"14",
+         "spark.yarn.executor.memoryOverhead":"1024",
+         "spark.dynamicAllocation.enabled":"false",
+         "spark.driver.extraJavaOptions":"-XX:+UseG1GC -XX:+UnlockDiagnosticVMOptions -XX:+G1SummarizeConcMark -XX:InitiatingHeapOccupancyPercent=35 -verbose:gc -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:OnOutOfMemoryError='kill -9 %p'"
```
Comment (wellfabr) [inline projects/orders-partition-historic/script-submit/config.json]: Same here for UseParallelGC vs UseG1GC

File: projects/orders-partition-historic/script-submit/config.json (hunk 1 above)
Comment (wellfabr) [reply projects/orders-partition-historic/script-submit/config.json]: The best way that worked for me to remove out of memory errors, in general, was keeping an eye on the transformations to see how the data is partitioned. The data should be well distributed across the partitions and you must have enough partitions (very big partitions may cause memory errors).

File: projects/orders-partition-historic/script-submit/config.json (hunk 2 above)
Comment (italo-batista) [reply projects/orders-partition-historic/script-submit/config.json]: What you have said about UseG1GC makes sense according to what we have realized on the execution of our jobs. Thanks.

File: projects/orders-partition-historic/script-submit/config.json (hunk 1 above)
Comment (italo-batista) [reply projects/orders-partition-historic/script-submit/config.json]: Thank you @wellfabr for reviewing these confs. We used them as committed above according to some articles we've found. But I registered your suggestions and we will try them in the future.

**Pull request merged**